src/library/FIFO.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A First-In-First-Out (FIFO) class that maintains a queue of
+ * elements.  New elements are added to the tail, and the oldest
+ * element is removed from the head when the FIFO is full.  If the
+ * size of the queue is 1, it will always contain the newest element
+ * that was inserted.
+ * @class
+ */
+
+class FIFO {
+  
+  /**
+   * @constructor Constructs a FIFO (First-In-First-Out) object.
+   * @param {Number} size The size of the FIFO.
+   */
+  constructor(size) {
+    /** @private @const {!size} The size of the FIFO */
+    this.size_ = size;
+    /** @private @const {!data} The buffer data of the FIFO */
```

Comment:
Perhaps:

"The actual data array for the FIFO."

The term "buffer" is pretty loaded in the audio programming. I wouldn't use it unless it contains playable PCM audio data.